src/dune_rules/install_rules.ml

```diff
@@ -160,26 +161,32 @@ end = struct
               | Some f -> [ (cm_kind, f) ]
             else []
           in
-          let open Cm_kind in
-          [ if_ true (Cmi, cm_file Cmi)
-          ; if_ native (Cmx, cm_file Cmx)
-          ; if_ (byte && virtual_library) (Cmo, cm_file Cmo)
+          let open Lib_mode.Cm_kind in
+          [ if_ (byte || native) (Ocaml Cmi, cm_file (Ocaml Cmi))
+          ; if_ melange (Melange Cmi, cm_file (Melange Cmi))
```

Comment:
For now, I would drop the installation of melange artifacts. You have no use for this feature yet, and this change alone isn't enough to make it work (you need symmetrical changes to read this stuff from dune-package).

Reply:
Fixed all this in https://github.com/jchavarri/dune/pull/4/commits/9f91d01c041930185c650e4a5e02cba2fcab8edf.